### PR TITLE
feat(anta.tests): Add known EOS errors as failure

### LINKED ISSUE #952: Fix: VerifyIPProxyARP should not throw an error
```
 ERROR    [anta.device] Command 'show ip interface Ethernet1' failed on spine1: Ethernet1 does not support IP  
```

### LINKED ISSUE #954: Fix: `VerifyReachability`
```
           ERROR    Command 'ping vrf MGMT 1.1.1.1 source Management0 size 100 df-bit repeat 2' failed on spine2: VRF 'MGMT' is not active                                                                                                           device.py:418

```

### PR DESCRIPTION
**Pull request by gmuloc**
# Description

Add a list of known errors from EOS to match on to decide a test status

Fixes #952 
Fixes #954 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have run pre-commit for code linting and typing (`pre-commit run`)
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

**Comment by carl-baillargeon** [comment]
Routing tests are also failing if the routing protocol is not active (BGP, IS-IS):
`DC2-BL2 :: VerifyEVPNType2Route :: ERROR(show bgp evpn route-type mac-ip aa:c1:ab:de:50:ad vni 10010 has failed: BGP inactive`

**Comment by gmuloc** [comment]
> show bgp evpn route-type mac-ip aa:c1:ab:de:50:ad vni 10010

do you have an example for ISIS?

File: anta/device.py [hunk 1]
```diff
@@ -386,6 +386,7 @@ async def _collect(self, command: AntaCommand, *, collection_id: str | None = No
         collection_id
             An identifier used to build the eAPI request ID.
         """
+        # pylint: disable=too-many-branches
```
Comment (carl-baillargeon) [inline anta/device.py:389]: Yeah at some point we'll have to split up this function.

Comment (carl-baillargeon) [reply anta/device.py:389]: Maybe in this PR? I think having a `_handle_eapi_command_error` method should be enough to avoid ruff/pylint complaining.

Comment (gmuloc) [reply anta/device.py:389]: do you want "some point" to be now? :p

Comment (gmuloc) [reply anta/device.py:389]: yeah will try to update here - hopefully not too many new unit tests to write!

File: anta/models.py [hunk 2]
```diff
@@ -250,8 +255,23 @@ def supported(self) -> bool:
...
+        """Return True if the command returned a known_eos_error on the device, False otherwise.
+
+        RuntimeError
+            If the command has not been collected and has not returned an error.
+            AntaDevice.collect() must be called before this property.
+        """
+        if not self.collected and not self.error:
+            msg = f"Command '{self.command}' has not been collected and has not returned an error. Call AntaDevice.collect()."
+            raise RuntimeError(msg)
+        known_eos_errors = ["BGP inactive", "VRF '.*' is not active", ".* does not support IP"]
```
Comment (carl-baillargeon) [inline anta/models.py:273]: Should we put this in a separate module? Might be easier to maintain/access if we need to add stuff later.

Comment (gmuloc) [reply anta/models.py:273]: I vote when the time comes we do it :p

**Comment by carl-baillargeon** [comment]
> > show bgp evpn route-type mac-ip aa:c1:ab:de:50:ad vni 10010
> 
> do you have an example for ISIS?

@gmuloc:
```
{
    "vrfs": {
        "default": {
            "isisInstances": {}
        }
    },
    "errors": [
        "IS-IS (IS-IS) is disabled because: IS-IS address family configuration is not present"
    ]
}
```
It happens when you don't have address-family ipv4 | ipv6 configured

File: anta/models.py (hunk 2 above)
Comment (carl-baillargeon) [reply anta/models.py:273]: The time has come! (IS-IS) :P

**Comment by gmuloc** [comment]
> Member
> ###

what is the command please? need to repro

**Comment by gmuloc** [comment]
ok found one :p

File: anta/models.py [hunk 3]
```diff
@@ -34,6 +34,14 @@
 
 logger = logging.getLogger(__name__)
 
+KNOWN_EOS_ERRORS = [
```
Comment (carl-baillargeon) [inline anta/models.py:37]: We have `anta.constants` module.

File: anta/device.py [hunk 4]
```diff
@@ -446,6 +438,18 @@ async def _collect(self, command: AntaCommand, *, collection_id: str | None = No
...
 
+    def _log_eapi_command_error(self, command: AntaCommand, e: asynceapi.EapiCommandError) -> None:
+        """Appropriately log the eapi command error."""
+        command.errors = e.errors
+        if command.requires_privileges:
+            logger.error("Command '%s' requires privileged mode on %s. Verify user permissions and if the `enable` option is required.", command.command, self.name)
+        if not command.supported:
+            logger.debug("Command '%s' is not supported on '%s' (%s)", command.command, self.name, self.hw_model)
+        elif command.returned_known_eos_error:
+            logger.debug("Command '%s' returned a known error '%s'", command.command, self.name)
```
Comment (carl-baillargeon) [inline anta/device.py:449]: Not sure if you agree but to me these debug logs are irrelevant:
image

debug logs should provide more information and context versus regular logs. That's actually the opposite here, we have less information in debug logs :P

File: anta/models.py [hunk 5]
```diff
@@ -34,6 +34,14 @@
...
+    r"BGP inactive",
+    r"VRF '.*' is not active",
+    r".* does not support IP",
```
Comment (carl-baillargeon) [inline anta/models.py:40]: This was for VerifyReachability right? If so, I am still getting an error:
`ERROR    [anta.device] Command 'ping vrf MGMT 1.1.1.1 source Management1 size 100 df-bit repeat 2' failed on DC1-LEAF1A: No source interface Management1 `

This happens with an interface not configured on the switch. Another example:
`ERROR    [anta.device] Command 'ping vrf MGMT 1.1.1.1 source Ethernet100 size 100 df-bit repeat 2' failed on DC1-LEAF1A: No source interface Ethernet100 `

File: anta/models.py [hunk 6]
```diff
@@ -657,6 +679,30 @@ async def wrapper(
...
+        unsupported_commands = [f"'{c.command}' is not supported on {self.device.hw_model}" for c in cmds if not c.supported]
+        if unsupported_commands:
+            msg = f"Test {self.name} has been skipped because it is not supported on {self.device.hw_model}: {GITHUB_SUGGESTION}"
+            self.logger.warning(msg)
+            self.result.is_skipped("\n".join(unsupported_commands))
+            return
+        returned_known_eos_error = [f"'{c.command}' failed on {self.device.name}: {', '.join(c.errors)}" for c in cmds if c.returned_known_eos_error]
+        if returned_known_eos_error:
+            msg = f"Test {self.name} has failed."
+            self.logger.warning(msg)
```
Comment (carl-baillargeon) [inline anta/models.py:700]: `WARNING  [anta.tests.routing.isis.VerifyISISNeighborState] Test VerifyISISNeighborState has failed.`

The test will be marked as failed in the report so here users might ask themselves "Why am I getting a log for this failed test but not the others?".

I think we should either remove this log or provide more context/details just like we do for `unsupported_commands` which provides more context why it was skipped than the actual message in the report.

File: pyproject.toml [hunk 7]
```diff
@@ -70,7 +70,6 @@ dev = [
   "pytest-cov>=4.1.0",
   "pytest-dependency",
   "pytest-codspeed>=2.2.0",
-  "respx",
```
Comment (carl-baillargeon) [inline pyproject.toml:73]: No bueno :)

File: anta/models.py (hunk 5 above)
Comment (gmuloc) [reply anta/models.py:40]: > r VerifyReachability  error was `r"VRF '.*' is not active",` if you look at theissue

yours is new :p we need to add it but it was never reported

File: pyproject.toml (hunk 7 above)
Comment (gmuloc) [reply pyproject.toml:73]: why is it there?

File: anta/models.py (hunk 3 above)
Comment (gmuloc) [reply anta/models.py:37]: not sure it helps honestly but done

File: anta/device.py (hunk 4 above)
Comment (gmuloc) [reply anta/device.py:449]: I dont think regular logs bring more context here. Maybe you meant to say the error message?

File: anta/models.py (hunk 6 above)
Comment (gmuloc) [reply anta/models.py:700]: I vote no log - the test has failed and that's enough

**Review by carl-baillargeon — APPROVED** [review]
Ship it!

**Pull request merged**